ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -288,27 +332,56 @@ private TransactionSelectionResult evaluateTransaction(
       return handleTransactionNotSelected(evaluationContext, selectionResult);
     }
 
-    final WorldUpdater txWorldStateUpdater = blockWorldStateUpdater.updater();
     final TransactionProcessingResult processingResult =
-        processTransaction(evaluationContext.getTransaction(), txWorldStateUpdater);
+        processTransaction(evaluationContext.getTransaction());
+
+    txWorldStateUpdater.markTransactionBoundary();
```

Comment:
Could you elaborate on why this is needed ?

Reply:
this is needed in case more that one tx is processed before the commit, for example a plugin can only decide to commit only if txA and txB are both selected, so in this case you use the same world state updater first to execute txA and then txB, but without marking the tx boundary, it is possible that when executing txB there are some addresses already warm-up by txA, thus breaking the gas calculation.
This has no effect without plugin, since here we are committing after every selected, but provides flexibility for plugin to implement selection logic.